### PR TITLE
refactor: support special characters in table keys

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR supports special characters in table keys.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by shuiyisong** [comment]
[Prometheus metric naming regex](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels)

Accepting `()'"` in table name feels a bit wired 🥲

**Comment by v0y4g3r** [comment]
> [Prometheus metric naming regex](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) only accepts `[a-zA-Z_:][a-zA-Z0-9_:]*`
> 
> Accepting `()'"` in table name feels a bit wired 🥲

@fengjiachun Do you think it's necessary to also accept `()'"`?

**Review by shuiyisong — APPROVED** [review]

File: src/catalog/src/helper.rs [hunk 1]
```diff
@@ -30,6 +30,7 @@ pub const TABLE_GLOBAL_KEY_PREFIX: &str = "__tg";
 pub const TABLE_REGIONAL_KEY_PREFIX: &str = "__tr";
 
 const ALPHANUMERICS_NAME_PATTERN: &str = "[a-zA-Z_][a-zA-Z0-9_]*";
+const TABLE_NAME_PATTERN: &str = "[a-zA-Z_][a-zA-Z0-9_:]*";
```
Comment (fengjiachun) [inline src/catalog/src/helper.rs:33]: Prometheus metric's first character can accept  ':'.

Comment (fengjiachun) [reply src/catalog/src/helper.rs:33]: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

**Review by fengjiachun — APPROVED** [review]
LGTM

**Comment by sunng87** [comment]
Can we include a test for accessing tables with special name like this?

**Pull request merged**